coderd/database/migrations/000289_agent_resource_monitors.up.sql

```diff
@@ -0,0 +1,16 @@
+CREATE TABLE workspace_agent_memory_resource_monitors (
+	agent_id 	uuid NOT NULL REFERENCES workspace_agents(id) ON DELETE CASCADE,
+	enabled 	boolean 					NOT NULL,
+	threshold 	integer 					NOT NULL,
+	created_at 	timestamp with time zone 	NOT NULL,
+	PRIMARY KEY (agent_id)
```

Comment:
Hhmm, I don't think this will work? The primary key needs to be unique.

https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-PRIMARY-KEYS

> A primary key constraint indicates that a column, or group of columns, can be used as a **unique identifier** for rows in the table.

---

In general, when you're creating database migrations, it's good practice to perform a few basic operations against them to validate that your design works.

I see what you're trying to do, but I think you're try to be overly efficient.

Reply:
Can you explain to me how we can have duplicates here ? I maybe am missing some logic. 

As we authorize only one resource monitor of type `memory` here - and the resource monitor is linked to an agent id - I feel like it is doing what we want ?

If this one breaks it means that there is multiple memory resource monitors declared in the terraform configuration.

Reply:
Hhmm, fair point. It's true that each agent should only have one memory resource monitor.

Please leave a comment in the migration explaining this constraint for future readers without context 👍 
_I should not review PRs before bed 🫤_